src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -23,6 +23,23 @@ public sealed class FormatHex : PSCmdlet
     {
```

Comment:
@vexx32, your last commit had 3 failures in `PowerShell-CI-linux`
FormatHex.Encoding Parameter.Can process UTF32 encoding 'fhx -InputObject 'hello' -Encoding UTF32'
```powershell
Expected 1, but got 2.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 373
373:             $result.count | Should -Be $Count
```
FormatHex.Cmdlet Functionality.Validate that streamed text does not have buffer underrun problems ''a' * 30 | Format-Hex'
```powershell
Expected the value to have type [Microsoft.PowerShell.Commands.ByteCollection] or any of its subtypes, but got @(00000000000000000000   61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61  aaaaaaaaaaaaaaaa, 00000000000000000010   61 61 61 61 61 61 61 61 61 61 61 61 61 61        aaaaaaaaaaaaaa  ) with type [System.Object[]].
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 485
485:         , $result | Should -BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
```
FormatHex.Count and Offset parameters.Offset = length
```powershell
Expected an empty collection, but got collection with size 1 @().
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 524
524:         $result.Bytes | Should -HaveCount 0
```